homeassistant/components/mysensors/config_flow.py

```diff
@@ -135,18 +135,26 @@ async def async_step_user(self, user_input: dict[str, str] | None = None):
         """Create a config entry from frontend user input."""
         schema = {vol.Required(CONF_GATEWAY_TYPE): vol.In(CONF_GATEWAY_TYPE_ALL)}
         schema = vol.Schema(schema)
+        errors = {}
 
         if user_input is not None:
             gw_type = self._gw_type = user_input[CONF_GATEWAY_TYPE]
             input_pass = user_input if CONF_DEVICE in user_input else None
             if gw_type == CONF_GATEWAY_TYPE_MQTT:
-                return await self.async_step_gw_mqtt(input_pass)
+                mqtt_entries = self.hass.config_entries.async_entries("mqtt")
```

Comment:
Isn't it enough to just check if MQTT is loaded? Or maybe we should have an `is_connected` MQTT loader ?

Reply:
The mqtt integration can be setup and then then the config entry can be removed. So just checking `hass.config.components` isn't enough, I think. Were you thinking of something else?

Reply:
well sure, but that is quite the edge case?

Reply:
Why? That's how you remove the integration.

Reply:
The existence of a config entry is also not going to mean it's connected. I suggest you add `is_connected` to the MQTT integration like we did for cloud: https://github.com/home-assistant/core/blob/dev/homeassistant/components/cloud/__init__.py#L119-L123

Reply:
We don't want to leak MQTT implementation details into the MySensors integration.

Reply:
Connected and setup are different things. It's ok to not be connected as long as we can connect without user intervention. It's not ok to not be setup.

Thr config entry state is not an integration detail. It's core state.

I can add a helper to check for mqtt connection state but that's not what this PR is aiming to solve.

Reply:
The fact that the connection is represented as a config entry could be seen as an implementation detail.

Reply:
Ok, I'll change to just check `hass.config.components`.